modules/UI/shared_video/SharedVideo.js

```diff
@@ -421,20 +438,63 @@ export default class SharedVideoManager {
      * @param {boolean} indicates if this mute was a result of user interaction,
      * i.e. pressing the mute button or it was programatically triggerred
      */
-    localAudioMuted (muted, userInteraction) {
+    onLocalAudioMuted (muted, userInteraction) {
         if(!this.player)
             return;
 
         if (muted) {
             this.mutedWithUserInteraction = userInteraction;
-            return;
         }
+        else if (this.player.getPlayerState() !== YT.PlayerState.PAUSED) {
+            this.mutePlayer(true);
+        }
+    }
+
+    /**
+     * Mutes / unmutes the player.
+     * @param mute true to mute the shared video, false - otherwise.
+     */
+    mutePlayer(mute) {
+        if (!this.player.isMuted() && mute) {
+            this.player.mute();
+            this.smartUnmute();
```

Comment:
As we discussed a little bit off-site, we need this in order to keep the volume untouched when the user mutes the video and then restore the volume once it's unmuted. I think I've taken care of the handling and Damyan agreed that it looks good for now.